### PR TITLE
Test tx decoding via fixture

### LINKED ISSUE #3111: Add CI check for breaking the transaction format
Add CI check to make sure that we don't break the transaction serialization format by accident

### PR DESCRIPTION
**Pull request by Fraccaman**
## Describe your changes
Closes #3111
## Indicate on which release or other PRs this topic is based on
#3130 
## Checklist before merging to `draft`
- [ ] I have added a changelog
- [ ] Git history is in acceptable state

### CODE REVIEW

File: examples/generate_txs.rs [hunk 1]
```diff
@@ -27,10 +28,11 @@ async fn main() -> Result<(), Reason> {
         sechash_map.insert(0xff, HEXLOWER.encode(&tx.raw_header_hash().0));
         ledger_vector.name = format!("{}_{}", i, ledger_vector.name);
         test_vectors.push(ledger_vector.clone());
-        debug_vectors.push((ledger_vector, tx, tx_data, sechash_map));
+        debug_vectors.push((ledger_vector, tx.clone(), tx_data, sechash_map));
+        serialized_txs.push(HEXLOWER.encode(tx.to_bytes().as_ref()));
     }
     let args: Vec<_> = std::env::args().collect();
-    if args.len() < 3 {
+    if args.len() < 4 {
```
Comment (tzemanovic) [inline examples/generate_txs.rs:35]:
```suggestion
    if args.len() < 3 {
```
let's make the new last arg optional

File: examples/generate_txs.rs [hunk 2]
```diff
@@ -27,10 +28,11 @@ async fn main() -> Result<(), Reason> {
...
         eprintln!("Usage: generate-txs <vectors.json> <debugs.txt>");
```
Comment (tzemanovic) [inline examples/generate_txs.rs:36]:
```suggestion
        eprintln!("Usage: generate-txs <vectors.json> <debugs.txt> <txs.json (optional)>");
```

File: examples/generate_txs.rs [hunk 3]
```diff
@@ -39,5 +41,7 @@ async fn main() -> Result<(), Reason> {
     std::fs::write(&args[1], json).expect("unable to save test vectors");
     std::fs::write(&args[2], format!("{:#?}", debug_vectors))
         .expect("unable to save test vectors");
+    std::fs::write(&args[3], format!("{:#?}", serialized_txs))
```
Comment (tzemanovic) [inline examples/generate_txs.rs:44]:
```suggestion
    std::fs::write(&args[3], serde::json::to_string(&serialized_txs).unwrap())
```

**Review by tzemanovic — APPROVED** [review]

**Comment by brentstone** [comment]
can you add changelog pls?

**Comment by tzemanovic** [comment]
> can you add changelog pls?

we don't require it for tests only changes

**Pull request merged**